winpr/libwinpr/crt/unicode_icu.c

```diff
@@ -61,7 +61,7 @@ int int_MultiByteToWideChar(UINT CodePage, DWORD dwFlags, LPCSTR lpMultiByteStr,
 		len = cbMultiByte;
 
 	if (len >= INT_MAX)
-		return -1;
+		return 0;
```

Comment:
I´d consider an invalid input argument an error and not silently ignore that

Reply:
I did not find doc for this function (did not search a lot).
I aligned this with the other functions.
In  my oppinion, int_MultiByteToWideChar and int_WideCharToMultiByte, already tell they failed by returning 0.
0 should not be considered as a valid return value, as a buffer of size 0 does not exist.

Reply:
The windows counter part returns 0 on error
```The function returns 0 if it does not succeed. To get extended error information, the application can call GetLastError, which can return one of the following error codes:

ERROR_INSUFFICIENT_BUFFER. A supplied buffer size was not large enough, or it was incorrectly set to NULL.
ERROR_INVALID_FLAGS. The values supplied for flags were not valid.
ERROR_INVALID_PARAMETER. Any of the parameter values was invalid.
ERROR_NO_UNICODE_TRANSLATION. Invalid Unicode was found in a string.```
I think there is no legitimate case where this function should return 0 and not fail.
Every null terminated string are of size >=1.
And giving a buffer of size 0 as input is semantically dubious as you won't need to convert it.
It is considered as an invalid input in both windows API and WinPr:
```C
	if ((cbMultiByte == 0) || (cbMultiByte < -1))
		return 0;
```
There are no valid input of the function that should result in it returning 0 on success.

It is a good idea to set error on ERROR_INVALID_PARAMETER.
It is possible to arrive at len >= INT_MAX if the function is fed a null terminated string of 2G+, which I can see calling an invalid parameter.

Reply:
ok, nice summary. sounds good to return `0` and set `ERROR_INVALID_PARAMETER`